### PR TITLE
fixed #202

### LINKED ISSUE #202: ChannelInvoker.newPromise should pass file and line to eventLoop.newPromise
We normally want debugging information to keep track of where our promises are allocated in debug modes. However, if someone does `Channel.newPromise` instead of `eventLoop.newPromise`, that information is lost. The core issue is here: https://github.com/apple/swift-nio/blob/fef728cab0cdb65459e3028d7ab7ffece0f34f6f/Sources/NIO/ChannelInvoker.swift#L182-L184

This indirection should preserve the #file and #line specific debugging logic we use elsewhere.

### PR DESCRIPTION
**Pull request by tib**
Introduced some new parameters.

### Motivation:

I just wanted to fix #202 :)

### Modifications:

Created new parameters for the file and line newPromise method inside ChannelOutboundInvoker.

### Result:

File and line numbers are now passed to the eventLoop object, all unit tests have been passed.

### CODE REVIEW

File: Sources/NIO/ChannelInvoker.swift [hunk 1]
```diff
@@ -138,49 +138,49 @@ public protocol ChannelOutboundInvoker {
...
     public func triggerUserOutboundEvent(_ event: Any) -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         triggerUserOutboundEvent(event, promise: promise)
         return promise.futureResult
     }
 
-    private func newPromise() -> EventLoopPromise<Void> {
-        return eventLoop.newPromise()
+    public func newPromise<T>(file: StaticString = #file, line: UInt = #line) -> EventLoopPromise<T> {
```
Comment (normanmaurer) [inline Sources/NIO/ChannelInvoker.swift]: @tib can we keep this private for now ? Also I think we can just keep the old return type to remove some typing ;)

Comment (tib) [reply Sources/NIO/ChannelInvoker.swift]: Sure, I'll make the changes.

**Comment by normanmaurer** [comment]
@tib can you please amend your commit to follow our commit template for the message ? Basically what you added here as description in the issue :) Also please squash. Thanks!

**Comment by normanmaurer** [comment]
@swift-nio-bot test this please

**Comment by tib** [comment]
@normanmaurer I made the changes, now it should be fine.

**Comment by normanmaurer** [comment]
@tib awesome... will merge once CI finish

**Comment by normanmaurer** [comment]
@swift-nio-bot test this please

**Comment by normanmaurer** [comment]
Thanks again!

**Pull request merged**